src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1017,14 +1019,14 @@ private boolean needIndexOrdering()
     /**
      * Orders results when multiple keys are selected (using IN)
      */
-    private void orderResults(ResultSet cqlRows, QueryOptions options)
+    public void orderResults(ResultSet cqlRows, QueryOptions options)
     {
         if (cqlRows.size() == 0 || !needsPostQueryOrdering())
             return;
 
-        Comparator<List<ByteBuffer>> comparator = orderingComparator.prepareFor(table, options);
-        if (comparator != null)
-            Collections.sort(cqlRows.rows, comparator);
```

Comment:
It looks like the sort is removed here but not added somewhere else.

Reply:
I assumed we already have a test that exercises this, is that not the case?

Reply:
I assume we have a test for that too. I checked out the branch, and any relevant vector tests are failing right now due to an NPE when calling `restriction.findSupportingIndex`, which is before exercising the `postQuerySort` method. Is the goal to move sorting into the `postQuerySort` method?

Reply:
Hi @michaeljmarshall, thanks for checking that. I moved the sorting into the `postQuerySort` method via the DSU pattern as per [this commit](https://github.com/datastax/cassandra/pull/710/commits/22cf12dc806b99372f602ec7f7aff702fb727191).